### ISSUE TITLE
Panorama image not recognized when creating thumbnails

### ISSUE THREAD

**Issue by hupfdule** [photoview/photoview#440]
**Describe the bug**
It seems photoview chokes on creating thumbnails for panorama images.

**To Reproduce**
Try to scan a panorama photo. (I can provide an example file if you tell me how I should send it to you. Attaching it to github doesn't work, because it is too large.)

**Expected behavior**
The photo should be scanned successfully and be displayed in photoview

Instead an error appears in the logfile:

```
 2021/07/19 20:00:03 ERROR: Failed to begin database transaction: failed to process photo (/photos/panorama.jpg): could
 not create thumbnail cached image: EOF
```

**Your environment:**

 - Are you using Docker? [yes]
 - Version [e.g. 2.3.4]

**Additional context**
The image was taken with the panorama functionality of a Samsung Galaxy S8 cellphone. All other images viewers I tried didn't have a problem with this photo.

**Comment by CoderTobi** [comment]
I have the same problem. Photo View goes ahead an scans all the pictures in a folder, it stops and doesn't process the remaining pictures in the Folder. Interestingly, I think it still scans the pictures of other folders. All the pictures still show up in the timeline and the album view. But they don't have a thumbnail and pressing the info-button shows "could not get media by media_id and user_id from database: record not found" in the sidebar. A rescan doesn't help.

The picture on which my instance failed has the following properties:
Resolution: 6128 x 1664
Size: 3,22 MB
Camera: Samsung Galaxy M21 SM-M215F

I am using the newest version of Photo View:
2.3.13

Unfortunately,
 this sadly makes it nearly unusable.

![Screenshot 2022-08-31 213025](https://user-images.githubusercontent.com/77673526/187766189-33cb13ef-9077-4c5e-ae41-45f8aff61401.png)

![Screenshot 2022-08-31 213620](https://user-images.githubusercontent.com/77673526/187767227-0e33dc7e-baaf-4e7d-8954-325349de47cf.png)

**Comment by kkovaletp** [comment]
@hupfdule , @CoderTobi , @JOHNKIMBLE , if the originally reported here issue with the Panorama images scanning is still reproducible on your system with the latest release, please share an example of the image, causing the issue in the ZIP archive without a password (in this way we could make sure that the image file is sent "as is" and not modified by a hosting). You can share it using any cloud drive by uploading the ZIP there and generating a sharing URL for everyone (public access)

**Comment by curiousercreative** [comment]
This [100MB TIFF](https://drive.google.com/file/d/16vSjjCr50LgY4QgELheful_HGQVfLqHt/view?usp=sharing) and many others reproduce this error
>ERROR: Error scanning media for album (541) file (/photos/1202_0022.tif): process media database transaction: process media (/photos/1202_0022.tif): creating high-res cached image: image decoding: failed to decode image (/photos/1202_0022.tif): unexpected EOF

**Comment by kkovaletp** [comment]
@curiousercreative, thanks for sharing the example. I confirm that it cannot be processed by the latest release build and throws the error. However, I think that this is not a representative example of this issue, because it is not a panorama image and the image is corrupted:
Our released version uses Darktable and an internal GO library to process images and it is sensitive to even small inconsistencies in the processed images, as it doesn't have an auto-correction algorithm.
We've migrated to the ImageMagick tool in our `master` branch, which tries to auto-correct recoverable data inconsistencies and process images as far as it can, so I've scanned your image successfully by the latest `master` build and see it in the UI.

I hope that this issue is completely handled by this migration and all your panorama images can be successfully scanned by the latest `master` version. Please try it.

<To everyone here>:
If you still see this issue on the `master` version for an image, that is correctly viewable by another image viewer (which means that it is not totally corrupted), please share it here and I'll test it on my side. 
Until that, I'll close this issue as completed.

**Comment by curiousercreative** [comment]
@kkovaletp thanks for the follow up. Out of curiosity, are we universally moving away from darktable or just for specific file types?

**Comment by kkovaletp** [comment]
@curiousercreative, yes, we dropped the Darktable dependency completely and switched to ImageMagick, which is more powerful and flexible for our needs

**Comment by curiousercreative** [comment]
@kkovaletp anywhere I cab read more about this? Will embedded JPEG previews still be used for RAW masters or will RAW files have to be developed for previews?

**Comment by kkovaletp** [comment]
@curiousercreative, we don't have any documentation on this, so you can read only merged PRs (the code and conversations).
The image processing logic remains the same as before, but now (thanks to ImageMagick) Photoview supports more image formats
